src/test/java/com/foundation/salesforce/steps/LeadsSteps.java

```diff
@@ -0,0 +1,200 @@
+     */
+    @When("a user finds a lead by Id (.*)")
+    public void aUserFindsLeadById(String leadId) {
+        response = requestManager.get(EndPoints.LEAD_ENDPOINT + "/" + leadId);
+        response.prettyPrint();
+    }
+
+    /**
+     * Verifies response's status code.
+     *
+     * @param statusCode - Expected status code.
+     */
+    @Then("the status code is {int}")
+    public void theStatusCodeIs(int statusCode) {
+        json = response.then().statusCode(statusCode);
+        Assert.assertEquals(response.getStatusCode(), statusCode);
+    }
+
+    /**
+     * Verifies response's json body.
+     *
+     * @param bodyFields - Expected fields and values.
+     */
+    @And("response includes the following")
+    public void responseIncludesTheFollowing(Map<String, String> bodyFields) {
+        SoftAssert softAssert = new SoftAssert();
+        for (Map.Entry<String, String> field : bodyFields.entrySet()) {
+            softAssert.assertEquals(response.jsonPath().get(field.getKey()).toString(), field.getValue());
+        }
+        softAssert.assertAll();
+    }
+
+    /**
+     * Retrieves a summary of all leads.
+     */
+    @When("a user retrieves the summary for lead")
+    public void aUserRetrievesTheSummaryForLead() {
+        response = requestManager.get(EndPoints.LEAD_ENDPOINT);
+        response.prettyPrint();
+    }
+
+    /**
+     * Deletes a lead an existing by Id.
+     */
+    @When("a user deletes an existing lead by Id")
+    public void aUserDeletesExistingLeadById() {
+        response = requestManager.delete(EndPoints.LEAD_ENDPOINT + "/" + context.getLead().getId());
+        response.prettyPrint();
+    }
+
+    /**
+     * Deletes a lead by Id.
+     *
+     * @param leadId - lead's Id.
+     */
+    @When("a user deletes a lead by Id (.*)")
+    public void aUserDeletesALeadById(String leadId) {
+        response = requestManager.delete(EndPoints.LEAD_ENDPOINT + "/" + leadId);
+        response.prettyPrint();
+    }
+
+    /**
+     * Sets a json object according to input map.
+     *
+     * @param inputFields - Input data.
+     */
+    @Given("a user sets json object")
+    public void aUserSetsJsonObjectWithRequiredFields(Map<String, String> inputFields) {
+        requestManager.buildSpec(inputFields);
+    }
+
+    /**
+     * Creates a lead.
+     */
+    @When("the user creates the lead")
+    public void theUserCreatesTheLead() {
+        response = requestManager.post(EndPoints.LEAD_ENDPOINT);
+        response.prettyPrint();
+        context.setResponse(response);
+    }
+
+    /**
+     * Verifies response's body array of json objects.
+     *
+     * @param bodyFields - Expected fields and values.
+     */
+    @And("response contains the following")
+    public void responseContainsTheFollowing(Map<String, String> bodyFields) {
+        SoftAssert softAssert = new SoftAssert();
+        Map<String, String> responseFirstElement = response.jsonPath().getMap("[0]");
```

Comment:
I think you should use an expression instead of a number 0 in the getmap[0] position would make the code more understandable.

Reply:
It is used to refer to the first element of the array of json objects